tests/http/server/_base.py

```diff
@@ -98,3 +111,413 @@ def _test_disconnect(
                 self.assertEqual(code, expected_code)
                 self.assertEqual(request.code, expected_code)
                 self.assertEqual(body, expected_body)
+
+    @logcontext_clean
+    def _test_cancellation_at_every_await(
```

Comment:
I feel like this should be called `make_request_with_cancellation_test` or something